smoketest/scripts/cli/test_ha_virtual_server.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2021 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from base_vyostest_shim import VyOSUnitTestSHIM
+
+from vyos.configsession import ConfigSessionError
+from vyos.ifconfig.vrrp import VRRP
+from vyos.util import cmd
+from vyos.util import process_named_running
+from vyos.util import read_file
+from vyos.template import inc_ip
+
+PROCESS_NAME = 'keepalived'
+KEEPALIVED_CONF = VRRP.location['config']
+base_path = ['high-availability']
+
+vrrp_interface = 'eth1'
+
+def getConfig(string, end='}'):
+    command = f'cat {KEEPALIVED_CONF} | sed -n "/^{string}/,/^{end}/p"'
```

Comment:
Why are you shelling out here? That would be just a few lines to do in Python.

Reply:
Replaced with `with open(KEEPALIVED_CONF) as f:`

Reply:
Why not use `vyos.util.read_file()`?

Reply:
It depends on what we want to read.
Do we want to read a whole config file or just a section in config file which include configuration related virtual_server

Reply:
with open and read_file do exactly the same. The latter is only a wrapper where we could inject debug code and/or assime default behavior.

Reply:
Replaced with `read_file(KEEPALIVED_CONF)`